### PR TITLE
Trigger calibration on 54L even if LFXO is used.

### PR DESCRIPTION
**Pull request by martintv**
Trying to test some tuning of PLL on 54L.

### CODE REVIEW

File: subsys/mpsl/init/mpsl_init.c [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 LOG_MODULE_REGISTER(mpsl_init, CONFIG_MPSL_LOG_LEVEL);
 
-#if defined(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC) && !defined(CONFIG_SOC_SERIES_NRF54HX)
+#if defined(CONFIG_MPSL_CALIBRATION_PERIOD) && !defined(CONFIG_SOC_SERIES_NRF54HX)
```
Comment (hermabe) [inline subsys/mpsl/init/mpsl_init.c:34]: Is this correct? Wont the config always be defined? Should we add another config `CONFIG_MPSL_DO_CLOCK_CALIBRATION` or similar and use that instead? Then we could get rid of the `&& !defined(CONFIG_SOC_SERIES_NRF54HX)` everywhere as well.

Comment (martintv) [reply subsys/mpsl/init/mpsl_init.c:34]: yeah, your right... this is broken on other platforms than 54L now.

Comment (martintv) [reply subsys/mpsl/init/mpsl_init.c]: I fixed it by adding if around this... 
I think a nicer fix would be to update the KConfig upstream here https://github.com/zephyrproject-rtos/zephyr/blob/e90c58a5b1ca4c9b040eb343d9c10b8ba59f3beb/drivers/clock_control/Kconfig.nrf#L57

and make it clear there that the calibartion needs to be done on 54L even if LFRC isnt used too.
but for now maybe this is good enogh?

**Review by olivier-le-sage — APPROVED** [review]
approving in advance since I'm guessing the nrfxlib update is on its way

**Pull request closed without merging**